### DISCUSSION TITLE
1.5.2 is out - did I manage to get micronucleus upload tool working on all platforms?

### DISCUSSION THREAD

**Discussion by SpenceKonde** [SpenceKonde/ATTinyCore#536] [General]
Board manager installs weren't working with micronucleus on windows in 1.5.0 or 1.5.1, and probably didn't work on linux either. They should now work. Can anyone confirm that they work on linux?

**Comment by prandeamus** [comment]
I can confirm that a board manager update from 1.5.0 to 1.5.2 was detected and ran the Linux install scripts, but I have not had the chance to test them fully. I had noticed that the 1.5.1 release from last week wasn't being detected as an update so I presume this was the cause.I hope that helps. 64-bit Elementary OS, (Ubuntu derivative) on Intel hardware.

**Reply by prandeamus** [reply]
Further comments on this - all good. I have 1.5.2 installed on a 64-bit Linux OS. I've been able to upload sketches via Micronucleus on both the existing MH-ET Live88 board that I had, and a new one out of the box. Micronucleus can be updated via Micronucleus just fine.

Incidentally the "stock" board had a MN version that identified as 2.2 and had 6650 bytes free in flash for programs. Doing an upgrade moved it to 2.5 with 6780 bytes free (130 bytes of flash, yay). Obviously if I want the full 8K I'll need to use ISP but this is so convenient for my needs. And the options provided to change the "enter boot loader" options are pitched just right for me.

In short: good job, and thanks to all. (I like to spread good news when I have it)

**Comment by pcfreak1201** [comment]
Well,  "update" from 1.5.1 to 1.5.2 says:
`Index error: could not find referenced tool name=micronucleus version=2.5-azd1b packager=ATTinyCore`
After manual removing and reinstall, it uses the tool from @ArminJo (also installed core):
```
/home/stefan/.arduino15/packages/digistump/tools/micronucleus/2.5/micronucleus --no-ansi --run --timeout 60 /tmp/arduino_build_691120/Blink_Tiny88.ino.hex 
> Please plug in the device (will time out in 60 seconds) ... 
> Device search timed out!
```
... but doesn't upload. I quick check the .json and see:

```
 "packager": "ATTinyCore",
              "name": "micronucleus",
              "version": "2.5-azd1b"
            }
          ]
        }
      ],
      "tools": [
        {
          "name": "micronucleus",
          "version": "2.5-azd1",
```

Is that a mistyping **1b to 1** or have I done something wrong? Only my linux-mint machine?

**Comment by SpenceKonde** [comment]
No, it's because I added the definitions of the 1b version to the copy of the package index that I was testing locally, but didn't copy it back to the one that I uploaded.

1b is identical to 1 for linux, and same binaries but different directory structure and compression formats for windows. 

It should work now.

**Reply by pcfreak1201** [reply]
Confirmed: message is gone, upload with t88_MH, works, t85 as still detection problems - but it can come from poking around. I'll check it with another PC...

**Reply by pcfreak1201** [reply]
Well, I had to reprogram the bootloader, but now it works also with t85

**Comment by SpenceKonde** [comment]
1.5.1 want showing up because I pulled it as soon as problems were
reported,expecting to have 1.5.2 up within hours. Obviously I managed to
mKe a trivial task days.

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***

**Comment by ArminJo** [comment]
I can confirm that burning bootloader and blink program both works for a ATtiny85 digispark 👍 
But how can I choose the different bootloader flavors?

**Comment by SpenceKonde** [comment]
That is coming in 2 .0.0, it is not in yet.
I realized there were so many fixes in github that weren't in release that
I should do release first then the big changes
____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***
